cluster-autoscaler/go.mod

```diff
@@ -55,6 +55,7 @@ require (
 	k8s.io/legacy-cloud-providers v0.0.0
 	k8s.io/utils v0.0.0-20231127182322-b307cd553661
 	sigs.k8s.io/cloud-provider-azure v1.29.4
```

Comment:
This should likely be 1.30.x? (And we probably need to give some thought to what it should be in master ...)

Reply:
(discussed offline) will be handled separately